src/test/java/uk/gov/hmcts/reform/managecase/api/controller/NoticeOfChangeControllerTest.java

```diff
@@ -394,4 +420,189 @@ private void postCallShouldReturnBadRequestWithErrorMessage(
                 .andExpect(jsonPath("$.errors", hasItem(caseIdInvalid)));
         }
     }
+
+    @Nested
+    @DisplayName("POST /noc/check-noc-approval")
+    @SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.JUnitTestsShouldIncludeAssert", "PMD.ExcessiveImports"})
+    class PostCheckNoticeOfChangeApproval extends BaseMvcTest {
+
+        private CheckNoticeOfChangeApprovalRequest request;
+        private CallbackCaseDetails caseDetails;
+        private ChangeOrganisationRequest changeOrganisationRequest;
+
+        private static final String ENDPOINT_URL = "/noc" + CHECK_NOTICE_OF_CHANGE_APPROVAL_PATH;
+
+        @BeforeEach
+        void setUp() {
+            changeOrganisationRequest = ChangeOrganisationRequest.builder()
+                .organisationToAdd(new Organisation("123", "Org1"))
+                .organisationToRemove(new Organisation("789", "Org2"))
+                .caseRoleId("CaseRoleId")
+                .requestTimestamp(LocalDateTime.now())
+                .approvalStatus(null)
+                .build();
+
+            given(jacksonUtils.convertValue(any(JsonNode.class), any()))
+                .willReturn(changeOrganisationRequest);
+        }
+
+        @DisplayName("happy path test without mockMvc")
+        @Test
+        void directCallHappyPath() {
+            changeOrganisationRequest.setApprovalStatus("1");
+            caseDetails = new CallbackCaseDetails(CASE_ID, "Jurisdiction", "State", "CaseTypeId",
+                                          Map.of("changeOrganisationRequestField",
+                                                 objectMapper.convertValue(changeOrganisationRequest, JsonNode.class)));
+            request = new CheckNoticeOfChangeApprovalRequest(null, null, caseDetails);
+            NoticeOfChangeController controller =
+                new NoticeOfChangeController(service,
+                                             approvalService,
+                                             verifyNoCAnswersService,
+                                             requestNoticeOfChangeService,
+                                             jacksonUtils);
+
+            SubmitCallbackResponse response = controller.checkNoticeOfChangeApproval(request);
+
+            assertThat(response).isNotNull();
+            assertThat(response.getConfirmationBody()).isEqualTo(CHECK_NOC_APPROVAL_DONE);
+            assertThat(response.getConfirmationHeader()).isEqualTo(CHECK_NOC_APPROVAL_DONE);
+        }
```

Comment:
Please also verify that the correct down stream service `noticeOfChangeApprovalService` has been called for happy paths that are approved.  **and in the non-approved test verify that it has not been called.**